lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SamuraisController < ApplicationController
+
+  def show
+    render json: samurai.to_json(only: %w[id name armor battles_count join_date death_date clan_id])
+  end
+
+  def index
+    samurais = clan.samurais
+    render json: samurais.to_json(only: %w[id name armor battles_count join_date death_date clan_id])
+  end
+
+  def create
+    samurai = clan.samurais.create!(samurai_params)
+    render json: samurai.to_json(only: %w[id name armor battles_count join_date death_date clan_id]), status: :created
+  end
+
+  def update
+    samurai.update!(samurai_params)
+    render json: samurai.to_json(only: %w[id name armor battles_count join_date death_date clan_id])
+  end
+
+  def destroy
+    samurai.destroy!
+    head 	:no_content
+  end
+
+
+  def alive
```

Comment:
Najlepiej nie tworzyć takich `nierestowych` akcji, tylko obsługiwać je jakoś w odpowiedniej restowej akcji(dostępne w CRUD to show, index, create, update, destroy) albo wydzielić do nowego zagnieżdżonego kontrolera, w naszym przypadku 100% full super restowo ścieżki by wyglądały tak:

```ruby
resources :clans, param: :name  do
  resources :samurais do
    resources :alive, only: %i[index]
    resources :dead, only: %i[index]
  end
end
```

Oczywiście staramy się unikać zagnieżdżeń więc to póki co samo obsługiwanie 2 query params w indexie by wystarczyło, ale jeśli mielibyśmy np. inne akcje `update` dla żywych i martwych samurai, to taką architekturę dużo łatwiej się rozbudowuje i ładniej dzieli tą logikę :D